### PR TITLE
Eliminate test only methods on types.Block

### PR DESCRIPTION
**Pull request by rawfalafel**
Noticed that there were a couple of methods on `types.Block` that were only being used to facilitate testing. These methods shouldn't be exposed to prod, so I moved them into the test file.

### CODE REVIEW

File: beacon-chain/blockchain/core.go [hunk 1]
```diff
@@ -140,6 +140,10 @@ func (b *BeaconChain) CanProcessBlock(fetcher types.POWBlockFetcher, block *type
 		return false, err
 	}
 
+	if time.Now().Before(genesisTime.Add(slotDuration)) {
```
Comment (rawfalafel) [inline beacon-chain/blockchain/core.go]: this is slightly unrelated; moved the timestamp check up to where it belongs.

**Comment by rawfalafel** [comment]
I renamed `NewBlockWithData` to `NewBlock` here btw. The old `NewBlock` was never used in prod, and the new `NewBlock` is now the only way to create a new block outside of its own package.

**Comment by terenc3t** [comment]
Ok. I was just wondering if the old `NewBlock` was going to get used in prod, but just not yet. Either way, the new `NewBlock` covers it's functionality

**Comment by rawfalafel** [comment]
Oh I see. The proposer will want to create a new block, so they might want their own method. Most likely not what the old `NewBlock` provided.

**Pull request merged**